common/simulation/spi.cpp

```diff
@@ -22,11 +22,16 @@ void sim_spi::SimTMC2130Spi::transmit_receive(
     uint8_t reg = control & ~write_mask;
 
     if (control & write_mask) {
+        // This is a write command.
         register_map[reg] = data;
+    } else {
+        // A read command will return the data from the previous read command.
+        auto out_iter = receive.begin();
+        out_iter =
+            bit_utils::int_to_bytes(read_register, out_iter, receive.end());
+        out_iter = bit_utils::int_to_bytes(register_map[read_register],
```

Comment:
_Technically_ this should be returned even if the write_mask is enabled. This diagram from the datasheet does a good job of summarizing the behavior.
![image](https://user-images.githubusercontent.com/27798632/149008805-53d2102a-1092-4478-be19-167bd2619eab.png)

Reply:
Awesome. Thanks.